rvt-patch-llvm/src/main.rs

```diff
@@ -392,12 +392,36 @@ where
     None
 }
 
-fn handle_panic(module: &Module) {
+fn handle_panic(context: &Context, module: &Module) {
     // TODO: make "spanic" a CL arg.
     if let Some(spanic) = module.get_function("spanic") {
-        if let Some(unwind) = module.get_function("rust_begin_unwind") {
-            unwind.replace_all_uses_with(spanic);
-            info!("Replaced panic handling ('rust_begin_unwind') with 'spanic'.");
+        // Collect all the functions that start with
+        // "_ZN3std9panicking11begin_panic17h" (there will be more than one).
+        let mut panics = vec![];
+        let mut op_fun = module.get_first_function();
+        while let Some(fun) = op_fun {
+            if fun.get_name()
+                .to_str()
+                .map(|s| s.starts_with("_ZN3std9panicking11begin_panic17h"))
+                .unwrap_or(false)
+            {
+                panics.push(fun);
+            }
+            op_fun = fun.get_next_function();
+        }
+
+        // Delete the body of each panic function and replace it with a call to
+        // `spanic`.
+        let builder = context.create_builder();
```

Comment:
I think that this is almost exactly what stub_functions does.
Specifically, I would think of writing it like this:
```
stub_functions(&context, &module, &functions, |builder| { builder.build_call(spanic, &[], "call"); builder.build_return(None); })
```

The only problem is that the second argument (&functions) takes a list of function names as an argument.

Maybe stub_functions should be changed so that the third argument (&functions) is replaced with either
- a vector/iterator of FunctionValue
or
- a predicate to test function names "Fn(String) -> bool"